lib/Froxlor/Validate/Validate.php

```diff
@@ -62,32 +62,77 @@ public static function validate($str, $fieldname, $pattern = '', $lng = '', $emp
 	}
 
-	/**
-	 * Checks whether it is a valid ip
-	 *
-	 * @param string $ip
-	 *        	ip-address to check
-	 * @param bool $return_bool
-	 *        	whether to return bool or call \Froxlor\UI\Response::standard_error()
-	 * @param string $lng
-	 *        	index for error-message (if $return_bool is false)
-	 * @param bool $allow_localhost
-	 *        	whether to allow 127.0.0.1
-	 * @param bool $allow_priv
-	 *        	whether to allow private network addresses
-	 * @param bool $allow_cidr
-	 *        	whether to allow CIDR values e.g. 10.10.10.10/16
-	 *        	
-	 * @return string|bool ip address on success, false on failure
-	 */
-	public static function validate_ip2($ip, $return_bool = false, $lng = 'invalidip', $allow_localhost = false, $allow_priv = false, $allow_cidr = false, $throw_exception = false)
+    /**
+     * Converts CIDR to a netmask address
+     *
+     * @thx to https://stackoverflow.com/a/5711080/3020926
+     * @param string $cidr
+     *
+     * @return string
+     */
+    public static function cidr2NetmaskAddr ($cidr) {
+
+        $ta = substr ($cidr, strpos ($cidr, '/') + 1) * 1;
+        $netmask = str_split (str_pad (str_pad ('', $ta, '1'), 32, '0'), 8);
+
+        foreach ($netmask as &$element) {
+            $element = bindec ($element);
+        }
+
+        return implode ('.', $netmask);
+    }
+
+    /**
+     * Checks if an $address (IP) is IPv6
+     *
+     * @param $address
+     *
+     * @return bool
+     * @thx https://stackoverflow.com/a/13677565/3020926
+     */
+    public static function is_ipv6($address) {
+        $ipv4_mapped_ipv6 = strpos($address, "::ffff:");
+        return (strpos($address, ":") !== FALSE) &&
+            ($ipv4_mapped_ipv6 === FALSE || $ipv4_mapped_ipv6 != 0);
+    }
+
+    /**
+     * Checks whether it is a valid ip
+     *
+     * @param string $ip
+     *            ip-address to check
+     * @param bool   $return_bool
+     *            whether to return bool or call \Froxlor\UI\Response::standard_error()
+     * @param string $lng
+     *            index for error-message (if $return_bool is false)
+     * @param bool   $allow_localhost
+     *            whether to allow 127.0.0.1
+     * @param bool   $allow_priv
+     *            whether to allow private network addresses
+     * @param bool   $allow_cidr
+     *            whether to allow CIDR values e.g. 10.10.10.10/16
+     * @param bool   $cidr_as_netmask
+     *            whether to format CIDR nodation to netmask notation
+     *
+     * @return string|bool ip address on success, false on failure
+     */
+	public static function validate_ip2($ip, $return_bool = false, $lng = 'invalidip', $allow_localhost = false, $allow_priv = false, $allow_cidr = false, $throw_exception = false, $cidr_as_netmask = false)
 	{
 		$cidr = "";
 		if ($allow_cidr) {
 			$org_ip = $ip;
 			$ip_cidr = explode("/", $ip);
-			if (count($ip_cidr) == 2) {
+			if (count($ip_cidr) === 2) {
+			    if (self::is_ipv6($ip)) {
```

Comment:
you call somethin `is_ipv6()` and then validate whether it's ipv6 using `filter_var`....seems odd to me

Reply:
First I check if its any type of v6, then I check (with `ip_cidr`) if it contained `/` which means its a ipv6 cidr notation. Thats a no go here, ipv6 without notation is just fine. I dont see any issue here.

Reply:
well the "issue" is, why have a function `is_ipv6()` when there's `filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6)` - hence my question what's up with the `is_ipv6()` function

Reply:
Because `filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4)` reports false, because of the CIDR notation. And if ipv4 is checked with the IPv6 flag, guess what happens.

Reply:
Ugh? At that point, there's no need for the CIDR notation for validation, you're exploding `$ip_cidr` and can use `$ip_cidr[0]` or am I missing something?

Reply:
Brainfart by me. Removed the inner If-Statement. My bad.